### PR TITLE
Implemented basic funcitonality for Firebase Analytics

### PR DESCRIPTION
**Pull request by BrandonStalnaker**
## Summary
Implement Firebase Analytics functionality.

## Testing Plan
Tested on QA environment and in unit tests.

## Master Issue
Closes mParticle/issues/issues/3592

### CODE REVIEW

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#import <Foundation/Foundation.h>
+#if defined(__has_include) && __has_include(<mParticle_Apple_SDK/mParticle.h>)
+#import <mParticle_Apple_SDK/mParticle.h>
+#else
+#import "mParticle.h"
+#endif
+
+static NSString *const googleAppIDKey = @"firebaseAppId";
+static NSString *const senderID = @"googleProjectNumber";
+static NSString *const userIdFieldKey = @"userIdFielduserIdField";
```
Comment (samdozor) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h]: I think this should be `userIdField`

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    _configuration = configuration;
+    
+    if ([FIRApp defaultApp] == nil) {
+        static dispatch_once_t FirebasePredicate;
+        
+        dispatch_once(&FirebasePredicate, ^{
+            NSString *googleAppId = configuration[googleAppIDKey];
+            NSString *gcmSenderId = configuration[senderID];
+            
+            FIROptions *options = [[FIROptions alloc] initWithGoogleAppID:googleAppId GCMSenderID:gcmSenderId];
```
Comment (samdozor) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: can you add some validation that these are not nil or empty?

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+    return [self execStatus:MPKitReturnCodeSuccess];
+}
+
+- (id const)providerKitInstance {
+    return [self started] ? self : nil;
+}
+
+- (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
+    if ([FIRApp defaultApp] == nil) {
```
Comment (samdozor) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: this should be impossible - if the kit reports itself as started, then that means there is a valid firebase instance

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+                                             kFIRParameterTransactionID: commerceEvent.transactionAttributes.transactionId
+                                             }];
+        }
+            break;
+            
+        default:
+            break;
+    }
+    
+    return [self execStatus:MPKitReturnCodeSuccess];
```
Comment (samdozor) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: I think you should only return success if data was forwarded. So for impressions and promotions, and for any actions that aren't covered above, this should return failure.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+}
+
+- (MPKitExecStatus *)logScreen:(MPEvent *)event {
+    if (![FIRApp defaultApp] && self.firebaseOptions != nil) {
```
Comment (samdozor) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: same as above - shouldn't this be impossible? couple other cases of this below, too.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+    if ([FIRApp defaultApp] == nil) {
+        if (self.firebaseOptions != nil) {
+            [FIRApp configureWithOptions:self.firebaseOptions];
+        } else {
+            return [self execStatus:MPKitReturnCodeFail];
+        }
+    }
+    
+    [FIRAnalytics setUserID:[self userIdForFirebase:user]];
+    [self logUserAttributes:user.userAttributes];
```
Comment (samdozor) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: I think this pattern will end up missing a lot of user attributes, which are typically set after an identify/login/logout occurs, not while it occurs.

Most other kits use these methods to listen for user attribute changes: https://github.com/mParticle/mparticle-apple-sdk/blob/master/mParticle-Apple-SDK/Kits/MPKitProtocol.h#L87

**Review by samdozor — DISMISSED** [review]

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+                [FIRAnalytics logEventWithName:kFIREventAddToWishlist
+                                    parameters:parameters];
+            }
+        }
+            break;
+            
+        case MPCommerceEventActionCheckout: {
+            NSNumber *value = @0;
+            for (MPProduct *product in commerceEvent.products) {
+                value = @([value doubleValue] + [product.price doubleValue]);
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Should this be taking quantity into account?

File: Podfile [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+pod 'mParticle-Apple-SDK/mParticle', '~> 7.8.3'
+
+end
+
+target 'mParticle-Firebase-AnalyticsTests' do
+  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
+  # use_frameworks!
+
+  # Pods for mParticle-Firebase-AnalyticsTests
+pod 'Firebase/Core', '~> 5.16'
```
Comment (peterjenkins) [inline Podfile]: These version numbers can be removed to ease maintenance we just need them in the podspec

File: mParticle-Firebase-Analytics.podspec [hunk 9]
```diff
@@ -0,0 +1,23 @@
+Pod::Spec.new do |s|
+    s.name             = "mParticle-Firebase-Analytics"
+    s.version          = "7.6.0"
+    s.summary          = "-Firebase Analytics integration for mParticle"
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics.podspec]: Extra dash

File: mParticle-Firebase-Analytics.podspec [hunk 10]
```diff
@@ -0,0 +1,23 @@
+Pod::Spec.new do |s|
+    s.name             = "mParticle-Firebase-Analytics"
+    s.version          = "7.6.0"
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics.podspec]: This will be auto-updated but you might as well update to 7.8.0 if you get a chance just for hygiene

File: mParticle-Firebase-Analytics.podspec [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+    s.homepage         = "https://www.mparticle.com"
+    s.license          = { :type => 'Apache 2.0', :file => 'LICENSE' }
+    s.author           = { "mParticle" => "support@mparticle.com" }
+    s.source           = { :git => "https://github.com/mparticle-integrations/mparticle-apple-integration-firebase-analytics.git", :tag => s.version.to_s }
+    s.social_media_url = "https://twitter.com/mparticles"
+
+    s.ios.deployment_target = "8.0"
+    s.ios.source_files      = 'mParticle-Firebase-Analytics/*.{h,m,mm}'
+    s.ios.dependency 'mParticle-Apple-SDK/mParticle', '~> 7.8.3'
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics.podspec]: This will also be updated by CI but we don't specify the patch version, it's always 0.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h [hunk 12]
```diff
@@ -0,0 +1,37 @@
+//
+//  MPKitFirebaseAnalytics.h
+//
+//  Copyright 2018 mParticle, Inc.
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h]: This should be updated to 2019 maybe or just delete these comments? I defer to @samdozor obv not a big deal either way.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+//
+
+#import <Foundation/Foundation.h>
+#if defined(__has_include) && __has_include(<mParticle_Apple_SDK/mParticle.h>)
+#import <mParticle_Apple_SDK/mParticle.h>
+#else
+#import "mParticle.h"
+#endif
+
+static NSString *const googleAppIDKey = @"firebaseAppId";
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h]: Maybe worth renaming the variable name to match the value? Or not if you like it this way for some reason.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 14]
```diff
@@ -0,0 +1,384 @@
+//
+//  MPKitFirebaseAnalytics.m
+//
+//  Copyright 2018 mParticle, Inc.
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Same comment about copyright year or removal altogether or strip down to first three lines

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 15]
```diff
@@ -0,0 +1,384 @@
...
+
+@interface MPKitFirebaseAnalytics()
+
+@property (nonatomic, strong, readwrite) FIROptions *firebaseOptions;
+
+@end
+
+@implementation MPKitFirebaseAnalytics
+
+static NSString *const fbUserIdCustomerIDValue = @"customerId";
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Just noting that this might conflict if we end up having to build a facebook kit at some point, not sure what would be better though. Also not sure if we need to be prefixing things with kit prefixes or not, some kits do it and some don't, maybe to avoid potential linker errors? Not sure this is actionable but we could do something like kMPFIRUserIdValue or something.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 16]
```diff
@@ -0,0 +1,384 @@
...
+            dispatch_async(dispatch_get_main_queue(), ^{
+                NSDictionary *userInfo = @{mParticleKitInstanceKey:[[self class] kitCode]};
+                
+                [[NSNotificationCenter defaultCenter] postNotificationName:mParticleKitDidBecomeActiveNotification
+                                                                    object:nil
+                                                                  userInfo:userInfo];
+            });
+        });
+    } else {
+        _started = YES;
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Wondering if we ought to broadcast a notification here so `[MParticle sharedInstance] kitInstance:^(<...>) {}` will work properly.

We normally encourage customers to gate their direct usage of kit APIs based on the kit being active but if they're init'ing it themselves before they wait for our config that doesn't make sense.

It's a bit weird though to set the started variable to true without broadcasting the notification, if we changed core to KVO the kits and broadcast notifications on their behalf we wouldn't be able to preserve this behavior.

Again not sure if this is actionable feedback or not but core does listen to the notifications per the public facing API I mentioned above.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 17]
```diff
@@ -0,0 +1,384 @@
...
+                                                     kFIRParameterContentType: @"product",
+                                                     kFIRParameterItemID: product.sku
+                                                     }];
+                }
+            }
+        }
+            break;
+            
+        case MPCommerceEventActionViewDetail: {
+            if (commerceEvent.products.count > 1) {
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Just wanted to verify this is correct, we log a "view item list" with the first product and then "view item" for each subsequent?

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 18]
```diff
@@ -0,0 +1,384 @@
...
+                [FIRAnalytics logEventWithName:kFIREventViewItem
+                                    parameters:parameters];
+            }
+        }
+            break;
+            
+        case MPCommerceEventActionPurchase: {
+            NSNumber *value = @0;
+            for (MPProduct *product in commerceEvent.products) {
+                value = @([value doubleValue] + [product.price doubleValue]);
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: As above does this need to be taking quantity into account? Otherwise would potentially end up with incorrect revenue numbers, right?

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 19]
```diff
@@ -0,0 +1,384 @@
...
+            }
+            
+            parameters = [self getParameterForCommerceEvent:commerceEvent andProduct:nil withValue:value];
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Just wanted to confirm product is supposed to be nil here and not e.g. the first product, right?

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 20]
```diff
@@ -0,0 +1,384 @@
...
+
+            [FIRAnalytics logEventWithName:kFIREventEcommercePurchase
+                                parameters:parameters];
+        }
+            break;
+            
+        case MPCommerceEventActionRefund: {
+            NSNumber *value = @0;
+            for (MPProduct *product in commerceEvent.products) {
+                value = @([value doubleValue] + [product.price doubleValue]);
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Bit repetitive but wanted to flag this as another place that potentially needs quantity change

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 21]
```diff
@@ -0,0 +1,384 @@
...
+}
+
+- (MPKitExecStatus *)logScreen:(MPEvent *)event {
+    if (![FIRApp defaultApp] && self.firebaseOptions != nil) {
+        [FIRApp configureWithOptions:self.firebaseOptions];
+    } else {
+        return [self execStatus:MPKitReturnCodeFail];
+    }
+    
+    if (![FIRApp defaultApp] || !event || !event.name) return [self execStatus:MPKitReturnCodeFail];
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Not sure how we feel about single line if statements, I think the succinct nature of them but sometimes when debugging I've had to format them into multiline to get a proper breakpoint. Something we might want to address if we make a style guide at some point.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 22]
```diff
@@ -0,0 +1,384 @@
...
+    }
+    
+    NSArray *userAttributesKeys = userAttributes.allKeys;
+    for (NSString *attributeKey in userAttributesKeys) {
+        [FIRAnalytics setUserPropertyString:userAttributes[attributeKey] forName:attributeKey];
+    }
+}
+
+-(NSDictionary<NSString *, id> *)getParameterForCommerceEvent:(MPCommerceEvent *)commerceEvent andProduct:(MPProduct *)product withValue:(NSNumber *)value {
+    NSMutableDictionary<NSString *, id> *parameters = [[NSMutableDictionary alloc] init];
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: FWIW you can just use `[NSMutableDictionary dictionary]`.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m [hunk 23]
```diff
@@ -0,0 +1,384 @@
...
+        [parameters setObject:product.sku forKey:kFIRParameterItemID];
+    }
+    if (product.name) {
+        [parameters setObject:product.name forKey:kFIRParameterItemName];
+    }
+    if (product.category) {
+        [parameters setObject:product.category forKey:kFIRParameterItemCategory];
+    }
+    if (product.price) {
+        [parameters setObject:product.price forKey:kFIRParameterValue];
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Just wanted to confirm it's intended to set both value and price.

File: mParticle-Firebase-Analytics/mParticle-Firebase-Analytics.h [hunk 24]
```diff
@@ -0,0 +1,7 @@
+#import <UIKit/UIKit.h>
+
+FOUNDATION_EXPORT double mParticle-Firebase-AnalyticsVersionNumber;
+
+FOUNDATION_EXPORT const unsigned char mParticle-Firebase-AnalyticsVersionString[];
+
+#import <mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h>
```
Comment (peterjenkins) [inline mParticle-Firebase-Analytics/mParticle-Firebase-Analytics.h]: I think the other kits generally import using underscores not dashes.

File: mParticle-Firebase-AnalyticsTests/mParticle_Firebase_AnalyticsTests.m [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testStarted {
+    MPKitFirebaseAnalytics *exampleKit = [[MPKitFirebaseAnalytics alloc] init];
+    [exampleKit didFinishLaunchingWithConfiguration:@{googleAppIDKey:@"1:338209672096:ios:57235e7ff821ba85", senderID:@"338209672096"}];
```
Comment (peterjenkins) [inline mParticle-Firebase-AnalyticsTests/mParticle_Firebase_AnalyticsTests.m]: Not sure if these creds are ok to be public or not? Generally if a partner has given us creds we try to avoid exposing them where as if it's a self-signup situation they wouldn't be sensitive.

**Review by peterjenkins — CHANGES_REQUESTED** [review]
Mostly looks pretty good except possibly the quantity stuff, there's some minor potentially negligible feedback / questions in there but figured it'd be better to be risk raising a trivial issue rather than risk not commenting on something important.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 3 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: I'll do some more testing but I put this in here because of that strange scenario I showed you where [FIRApp defaultApp] is nil despite it being initialized with custom options previously.

Comment (samdozor) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: ok sounds good - please do. otherwise this is tech debt i'd like to avoid.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 18 above)
Comment (samdozor) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: i think actually it should be using the totalAmount/Revenue property of commerceevent as other kits do.

**Review by peterjenkins — DISMISSED** [review]
Didn't notice earlier but this is also missing a .travis.yml file. (The one from example repo is the latest.)

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 7 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Oooh good catch.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 23 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: yes but from your earlier comment I realized value should be taking quantity into account as well.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 7 above)
Comment (peterjenkins) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Just in case Sam's comment below got lost in the shuffle, apparently this should use a diff API vs. manually multiplying by quantity: https://github.com/mparticle-integrations/mparticle-apple-integration-google-analytics-firebase/pull/1#discussion_r254314866

File: mParticle-Firebase-AnalyticsTests/mParticle_Firebase_AnalyticsTests.m (hunk 25 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-AnalyticsTests/mParticle_Firebase_AnalyticsTests.m]: These are credentials I made using my mParticle email for a test project. The most information it could lead back to is my email.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h (hunk 13 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.h]: The field/variable is actually call google App ID in the googleInfo.plist, firebase console, and in FIROptions method below. Android and google internally just use this. So I figured keeping this would help make it more obvious that they represent the same thing.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 7 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: yea. Anytime I was calculating for the whole transaction before, i am now using transactionAttributes.revenue

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 17 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: If there's more than 1 product: we "view item list" with the first product representing the category of that view.
Then we log a view item for every product in that event, including that first one.

I could see changing this to do  either "view item list" if multiple (different) product OR "view item" is only one product but I felt like this way the client could decide how they want it done.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 19 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: Yep, we only pass product if the Parameters should only focus on a single product. If you leave it nil, the parameter builder creates parameters as if there's multiple products.

File: mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m (hunk 21 above)
Comment (BrandonStalnaker) [reply mParticle-Firebase-Analytics/MPKitFirebaseAnalytics.m]: My vote would be multiline. The number of lines matters more in stats than when actually looking at code.

File: mParticle-Google-Analytics-Firebase/MPKitFirebaseAnalytics.m [hunk 26]
```diff
@@ -0,0 +1,340 @@
...
+    if ([FIRApp defaultApp] == nil) {
+        static dispatch_once_t FirebasePredicate;
+        
+        dispatch_once(&FirebasePredicate, ^{
+            NSString *googleAppId = configuration[kMPFIRGoogleAppIDKey];
+            NSString *gcmSenderId = configuration[kMPFIRSenderIDKey];
+            
+            if (googleAppId && ![googleAppId isEqualToString:@""]) {
+                if (!gcmSenderId || [gcmSenderId isEqualToString:@""]) {
+                    NSLog(@"Invalid Google Project Number, %@. Using Default", googleAppId);
```
Comment (samdozor) [inline mParticle-Google-Analytics-Firebase/MPKitFirebaseAnalytics.m]: I would just return here rather than attempting to initialize with a known-bad gcmGenderId. Also - the error message need to be updated.

File: mParticle-Google-Analytics-Firebase/MPKitFirebaseAnalytics.m [hunk 27]
```diff
@@ -0,0 +1,340 @@
...
+- (NSString *)userIdForFirebase:(FilteredMParticleUser *)currentUser {
+    NSString *userId = currentUser.userId.stringValue;
+    if (currentUser != nil && self.configuration[kMPFIRUserIdFieldKey] != nil) {
+        NSString *key = self.configuration[kMPFIRUserIdFieldKey];
+        if ([key isEqualToString:kMPFIRUserIdValueCustomerID] && currentUser.userIdentities[@(MPUserIdentityCustomerId)] != nil) {
+            userId = currentUser.userIdentities[@(MPUserIdentityCustomerId)];
+        } else if ([key isEqualToString:kMPFIRUserIdValueEmail] && currentUser.userIdentities[@(MPUserIdentityEmail)] != nil) {
+            userId = currentUser.userIdentities[@(MPUserIdentityEmail)];
+        } else if ([key isEqualToString:kMPFIRUserIdValueMPID] && currentUser.userId != nil) {
+            userId = currentUser.userId != 0 ? [currentUser.userId stringValue] : @"0" ;
```
Comment (samdozor) [inline mParticle-Google-Analytics-Firebase/MPKitFirebaseAnalytics.m]: Can you return nil here, and then update the consumers of `userIdForFirebase` to handle nil (and just don't set the ID at all)? This will result in a hotkey/bad data downstream.

File: .travis.yml [hunk 28]
```diff
@@ -0,0 +1,9 @@
+osx_image: xcode10
+language: objective-c
+script:
+- sudo gem install cocoapods -v 1.5.3
+- travis_retry pod repo update > /dev/null
+- pod lib lint --use-libraries --allow-warnings || pod lib lint --allow-warnings
+- carthage update
+- carthage build --no-skip-current && carthage archive
+- xcodebuild -project mParticle-Example.xcodeproj -scheme mParticle_ExampleTests -destination 'platform=iOS Simulator,OS=12.0,name=iPhone XS' test
```
Comment (samdozor) [inline .travis.yml]: this line needs to be updated for naming.

**Review by samdozor — DISMISSED** [review]
